keyring/backends/Windows.py

```diff
@@ -100,7 +100,12 @@ def _set_password(self, target, username, password):
                           CredentialBlob=password,
                           Comment="Stored using python-keyring",
                           Persist=win32cred.CRED_PERSIST_ENTERPRISE)
-        win32cred.CredWrite(credential, 0)
+        try:
+            win32cred.CredWrite(credential, 0)
+        except pywintypes.error as e:
+            import sys
+            type, exc, tb = sys.exc_info()
+            raise PasswordSetError,("Max password length exceeded",type,exc),tb
```

Comment:
How do you know "password length exceeded" is the cause? Couldn't other causes have triggered a `pywintypes.error`? I'd like to see some test here and just raise the original exception if it isn't indicating a password of too great a length.